### PR TITLE
Report download progress on console in verbose mode

### PR DESCRIPTION
**Pull request by martyone**

### CODE REVIEW

File: src/libs/installer/packagemanagercore.cpp [hunk 1]
```diff
@@ -648,6 +648,20 @@ int PackageManagerCore::downloadNeededArchives(double partProgressSize)
...
         SIGNAL(progressChanged(double)), partProgressSize);
 
+    // Print some progress information on console as well
+    connect(&archivesJob, &DownloadArchivesJob::outputTextChanged, [](const QString &progress) {
+        qDebug().noquote() << progress;
+    });
+    connect(&archivesJob, &DownloadArchivesJob::progressChanged,
+            [lastReported = -10, time = QTime::currentTime()](double progress) mutable {
+        int roughProgress = static_cast<int>(progress * 10) * 10;
+        if (roughProgress > lastReported && time.elapsed() > 3000 || time.elapsed() > 15000) {
```
Comment (vigejolla) [inline src/libs/installer/packagemanagercore.cpp]: You could add parenthesis to improve readability:
`if ((roughtProgress > lastReported && time.elapsed() > 3000) || time.elapsed() > 15000)` is much faster to read and understand, as you don't have to think about operator precedence etc.

File: src/libs/installer/packagemanagercore.cpp [hunk 2]
```diff
@@ -648,6 +648,20 @@ int PackageManagerCore::downloadNeededArchives(double partProgressSize)
...
+            qDebug().nospace() << qMax(roughProgress, 1) << "% ...";
```
Comment (vigejolla) [inline src/libs/installer/packagemanagercore.cpp:657]: What's the point of qMax here? Why can't we report 0%?

File: src/libs/installer/packagemanagercore.cpp (hunk 1 above)
Comment (martyone) [reply src/libs/installer/packagemanagercore.cpp]: Think about && || operator precedence? :) Fixed

**Review by vigejolla — APPROVED** [review]
lgtm (even though I'm still waiting for the answer to the question regarding qMax)

File: src/libs/installer/packagemanagercore.cpp (hunk 2 above)
Comment (martyone) [reply src/libs/installer/packagemanagercore.cpp:657]: It would look odd (and less accurate)

**Pull request merged**